packages/embark-library-manager/src/npm.js

```diff
@@ -41,28 +70,37 @@ class Npm {
         this._logger,
         packageName,
         version,
-        'Downloading and installing {{packageName}} {{version}}...',
-        'Still downloading and installing {{packageName}} {{version}}... ({{duration}})',
-        'Finished downloading and installing {{packageName}} {{version}} in {{duration}}',
+        'Downloading and installing {{packageName}} {{version}}...\n',
+        'Still downloading and installing {{packageName}} {{version}}... ({{duration}})\n',
+        'Finished downloading and installing {{packageName}} {{version}} in {{duration}}\n',
         {
           showSpinner: !this._useDashboard,
           interval: this._useDashboard ? 2000 : 1000,
-          longRunningThreshold: 10000
+          longRunningThreshold: 600000
         }
       );
       timer.start();
 
-      // do the package download/install
-      pluginManager.install(packageName, version).then((result) => {
-        timer.end();
-        this._installing[packageName + version].forEach((cb) => {
-          cb(null, result.location);
+      // do the package install
+      const installPath = path.dirname(path.dirname(packagePath));
+      mkdirpSync(installPath);
+      exec(`npm init -y && npm install --no-package-lock --no-save ${packageName}@${version}`, {
+        cwd: installPath
+      }, (err) => {
+        this._installing[packageName + version].forEach(cb => {
+          setImmediate(() => { return err ? cb(err) : cb(null, packagePath); });
         });
         delete this._installing[packageName + version];
-      }).catch(err => {
-        this._installing[packageName + version].forEach((cb) => {
-          cb(err);
-        });
+
+        if (err) {
+          timer.abort();
+          this._installFailed[packageName + version] = err;
+          rimraf.sync(installPath);
+        } else {
+          timer.end();
+          this._installed[packageName + version] = true;
+          writeFileSync(path.join(installPath, 'finished'), '');
```

Comment:
@PascalPrecht here is where the "finished" comes into play.
We should probably put that in a constant above though.

Reply:
💯  Will switch to using a constant.